beacon-chain/core/epoch/epoch_operations.go

```diff
@@ -0,0 +1,338 @@
+package epoch
+
+import (
+	"bytes"
+	"fmt"
+
+	block "github.com/prysmaticlabs/prysm/beacon-chain/core/blocks"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/validators"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	b "github.com/prysmaticlabs/prysm/shared/bytes"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// Attestations returns the pending attestations of slots in the epoch
+// (state.slot-EPOCH_LENGTH...state.slot-1), not attestations that got
+// included in the chain during the epoch.
+//
+// Spec pseudocode definition:
+//   [a for a in state.latest_attestations
+//   if state.slot - EPOCH_LENGTH <= a.data.slot < state.slot]
```

Comment:
if then???

This definition should be more in the pattern of a list filter, in my opinion.

```python
return list(filter(lambda a: state.slot - EPOCH_LENGTH <= a.data.slot < state.slot, state.latest_attestations))
```

I guess that's not very readable, any `filter` would suffice, but this definition feels incomplete. I could be mistaken, my python is not very good. 

Consider adding something along the lines of 
```
Attestations returns the pending attestations where an attestation is considered to be pending when
state.slot - EPOCH_LENGTH <= attestation.data.slot < state.slot
```

Reply:
This was the old code that recently got reviewed and merged, and sure we can address it in this PR. I'll update the comments with a return